cdc/log.cc

```diff
@@ -1299,7 +1299,14 @@ cdc::cdc_service::impl::augment_mutation_call(lowres_clock::time_point timeout,
                 // iff a batch contains several modifications to the same table. Otoh, batch is rare(?)
                 // so this is premature.
                 tracing::trace(tr_state, "CDC: Selecting preimage for {}", m.decorated_key());
-                f = trans.pre_image_select(qs.get_client_state(), db::consistency_level::LOCAL_QUORUM, m);
+                f = trans.pre_image_select(qs.get_client_state(), db::consistency_level::LOCAL_QUORUM, m).then_wrapped([this] (future<lw_shared_ptr<cql3::untyped_result_set>> f) {
+                    auto& cdc_stats = _ctxt._proxy.get_cdc_stats();
+                    cdc_stats.counters_total.preimage_selects++;
+                    if (f.failed()) {
+                        cdc_stats.counters_failed.preimage_selects++;
+                    }
+                    return f;
```

Comment:
Does it make sense to return std::move(f) here or will the compiler complain about RVO being impossible then?

Reply:
This is the correct syntax.